### ISSUE TITLE
3.4.2.3.1 Completeness in EPUB A11Y 1.1

### ISSUE THREAD

**Issue by t-kudou** [mattgarrish/epub-revision#2221]
https://www.w3.org/TR/epub-a11y-11/#sec-mo-complete

Meeting this Objective
EPUB Creators MUST provide synchronized audio playback via Media Overlays for all textual content.

This statement could be interpreted as saying to provide synchronized audio using the Media Overlays mechanism, and not to use another mechanism.
Since the main purpose of this clause is to say that the complete audio narration should also be provided along with the full text, we believe that "for all textual content" is a weak explanation and should be "for complete textual content".

**Comment by murata2makoto** [comment]
Kudou-san is a member of the JDC technical committee.

Does the sentence quote by Kudou-san mandate the use of MO?  I guess not but it can be interpreted so.

Here is my interpretation of "via Media Overlays all textual content"

>If an element within the XHTML body element has child text content, this element MUST be referenced by some PAR element or MUST be a descendant of an element referenced by some PAR element.

**Comment by mattgarrish** [comment]
> Does the sentence quote by Kudou-san mandate the use of MO?

No, there is no requirement to use media overlays to meet the specification. And even if you use media overlays, you are not required to meet the objectives yet. We've kept that optional, as it says in the application section:

> To maximize the effectiveness of Media Overlays for people with different reading needs, however, EPUB Creators are strongly encouraged to meet the [OPTIONAL objectives](http://localhost:120/a11y/#sec-mo-obj) defined in the next section.

**Comment by mattgarrish** [comment]
> Here is my interpretation of "via Media Overlays all textual content"

It would have to account for SVG, too.

**Comment by murata2makoto** [comment]
> It would have to account for SVG, too.

Any `text ` element within an SVG content document MUST be referenced by some PAR element or MUST be a descendant of an element referenced by some PAR element.

**Comment by iherman** [comment]
It becomes a big more convoluted than that, @murata2makoto. It is not a `par` element that refers to a text element, it is a `text` SMIL element within a `par` that does that (via a `src` attribute).

Maybe it is enough to replace the reference to `par` to a `text`, and not refer to `par`; the structure of a valid SMIL document would require that `text` to be part of a `par` anyway.

**Comment by murata2makoto** [comment]
>It becomes a big more convoluted than that, @murata2makoto. It is not a par element that refers to a text element, it is a text SMIL element within a par that does that (via a src attribute).

Stand corrected.  My bad.

But how can we make sure that all text element within an SVG content document  is covered by SMIL?

**Comment by mattgarrish** [comment]
There's probably a larger question here about why we're referencing media overlays directly if we want this to be technology agnostic. If media overlays are replaced in a future epub, we have to rewrite all the objectives to account for the new technology.

The section should probably be "Synchronized Text-Audio Playback" and we should write out the specific references to media overlay documents and par/seq elements. That explanation belongs in the techniques.

**Comment by mattgarrish** [comment]
Assuming we move in this direction, the one good thing is that the techniques don't go to CR so we don't have to have any new additions in place before we transition. We'd just need to clean up the specification.
